test/test-find-node-directory.js

```diff
@@ -0,0 +1,115 @@
+var test = require('tape')
+var path = require('path')
+var findNodeDirectory = require('../lib/find-node-directory')
+
+var platforms = ['darwin', 'freebsd', 'linux', 'sunos', 'win32', 'aix']
+
+// we should find the directory based on the directory
+// the script is running in and it should match the layout
+// in a build tree where npm is installed in
+// .... /deps/npm
+test('test find-node-directory - node install', function (t) {
+  t.plan(platforms.length)
+  for (currentPlatform in platforms) {
+    var processObj = {execPath: '/x/y/bin/node', platform: currentPlatform}
+    t.equal(
+      findNodeDirectory('/x/deps/npm/node_modules/node-gyp/lib', processObj),
+                        path.join('/x'))
```

Comment:
without the join the patch ends up being incorrect on windows, the join fixes up the '/' to be appropriate for the platform